source/framework/vx_matrix.c

```diff
@@ -18,13 +18,73 @@
 
 #include <vx_internal.h>
 
-vx_matrix VX_API_CALL vxCreateMatrix(
-    vx_context context, vx_enum data_type, vx_size columns, vx_size rows)
+static vx_matrix ownCreateMatrix(vx_reference scope, vx_enum data_type, vx_size columns, vx_size rows, vx_bool is_virtual);
+static vx_status isMatrixCopyable(vx_matrix input, vx_matrix output);
+static vx_status VX_CALLBACK matrixKernelCallback(vx_enum kernel_enum, vx_bool validate_only, vx_enum optimization, const vx_reference params[], vx_uint32 num_params);
+
+/*! \brief This function is called to find out if it is OK to copy the input to the output.
+ * Columns, rows, data type, pattern, origin_x and origin_y must be the same
+ * \returns VX_SUCCESS if it is, otherwise another error code.
+ *
+ */
+static vx_status isMatrixCopyable(vx_matrix input, vx_matrix output)
```

Comment:
I have merged fixes for Matrix, Convolution, and Tensor meta data so you should be able to use tivxIsReferenceMetaFormatEqual for those now (please check).  Perhaps you can rebase your patches on top of main branch so you can take advantage of these fixes.